src/pipx/shared_libs.py

```diff
@@ -23,21 +23,39 @@
 
 class _SharedLibs:
     def __init__(self) -> None:
-        self.root = paths.ctx.shared_libs
```

Comment:
Why did you make this a property?

Reply:
This class is only instantiated exactly once: during loading of this module. (See https://github.com/pypa/pipx/blob/62aa16301edc70fb503adf5e54eaf48f955441d9/src/pipx/shared_libs.py#L151)

This is before any command line arguments are parsed and thus before e.g. the `--global` argument can have any effect. (Also before e.g. the monkeypatching in the test is happening.) This problem has previously already been (insufficiently) addressed [here](https://github.com/pypa/pipx/commit/863a3ec600da6b8dd5155d8b3a92a991f0244e50#diff-78cbc9ee1ca185b4d739dc2e7c8b263156522c3fe8f73ae4511b73755ed73139L49).

By converting all attributes to properties this problem is circumvented completely, because the paths are always recomputed based on the current `paths.ctx.shared_libs`. The necessary computations do not involve any file system operations and should therefore hopefully not be too expensive to repeat on every attribute-read.

PS: I do not particularly like that the `paths` and `shared_libs` use "statically" initialized global objects for exactly this reason. IMHO passing around immutable `paths` and `shared`_libs` objects as function parameters would simplify the whole thing and improve testability a lot; but it would pose a much bigger change than the present bugfix.

Reply:
Thanks for clarifying!

> PS: I do not particularly like that the `paths` and `shared_libs` use "statically" initialized global objects for exactly this reason. IMHO passing around immutable `paths` and `shared_libs` objects as function parameters would simplify the whole thing and improve testability a lot; but it would pose a much bigger change than the present bugfix.

Makes sense, let's put that on the todo list.